src/components/Navbar/components/NavbarRightList/NavbarRightList.tsx

```diff
@@ -1,20 +1,49 @@
-import Button from "../../../Button/Button"
-import { NavbarButton } from "../../Navbar.Styles"
+import {
+  NavbarButton,
+  NavbarToggleButton,
+} from "@/components/Navbar/Navbar.Styles"
 import * as S from "./NavbarRightList.Styles"
-import profile from "../../../../assets/profile.png"
-import { NAV_MENU_LIST } from "@/constants/stores"
+import profile from "@/assets/profile.png"
+import { useState } from "react"
+import NavbarLoggedInMenu from "./NavbarLoggedInMenu/NavbarLoggedInMenu"
+import NavbarNotLoggedInMenu from "./NavbarNotLoggedInMenu/NavbarNotLoggedInMenu"
+import MenuIcon from "@mui/icons-material/Menu"
+
+export type HandleMenuClickProps = (menuTitle: string) => void
 
 const NavbarRightList = () => {
+  // 나중에 전역 로그인 상태를 받아오도록 변경 예정
+  const [isLoggedIn, setIsLoggedIn] = useState(false)
+
+  const handleMenuClick: HandleMenuClickProps = (menuTitle) => {
+    if (menuTitle === "로그인") {
+      // navigate("/login")
+      handleLogin()
+    } else if (menuTitle === "로그아웃") {
+      // navigate("/")
+      // 전역에서 받은 로그인 상태가 true이면 로그아웃하고 홈으로 이동
+      handleLogout()
+    }
+  }
+
+  const handleLogin = (): void => {
+    setIsLoggedIn(true)
+  }
+  const handleLogout = (): void => {
+    setIsLoggedIn(false)
+  }
```

Comment:
훈오님이 저번에 리뷰해주신 코멘트가 생각나네요!! 
훈오님도 이렇게 해보시는 건 어떨지 제안드려봅니다! 🤔
```suggestion
const menuList = {
'로그인':handleLogin,
'로그아웃':handleLogout
}

  const handleMenuClick: HandleMenuClickProps = (menuTitle) => {
   menuList[menuTitle]()
  }
  
  const handleLogin = (): void => {
    setIsLoggedIn(true)
    // navigate('/login')
  }
  const handleLogout = (): void => {
    setIsLoggedIn(false)
    // navigate("/")
  }
```